### PR TITLE
@eessex => Fullscreen viewer

### PR DESCRIPTION
**Pull request by kanaabe**
fullscreen
TODO
- [x] Slide index
- [x] Imageset Preview link

This PR intentionally doesn't include the latest updates to images/artwork ViewFullscreen icon or the follow button within the slides. The size of this PR is getting a little too large so I'll leave those as a follow up!

### CODE REVIEW

**Comment by kanaabe** [comment]
hey @orta flagging this danger failure as possibly related to recent peril work?

**Comment by orta** [comment]
Thanks @kanaabe - looks like I need to do some more work on detecting the dangefile to run when dealing with cross-repo PRs, it's trying to use `master` on your repo instead of `slideshow`.

**Comment by orta** [comment]
Fixed, thanks

File: src/components/publishing/__stories__/images.story.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import Artwork from "../sections/artwork"
 import Image from "../sections/image"
 import ImageCollection from "../sections/image_collection"
 
-import { Artworks, Images } from "../__test__/fixtures/components"
+import { Artworks, Images } from "../fixtures/components"
```
Comment (eessex) [inline src/components/publishing/__stories__/images.story.tsx]: good call on file renaming

File: src/components/publishing/article.tsx [hunk 2]
```diff
@@ -1,33 +1,106 @@
...
 import FeatureLayout from "./layouts/feature_layout"
 import StandardLayout from "./layouts/standard_layout"
+import FullscreenViewer from "./sections/fullscreen_viewer/fullscreen_viewer"
 import Sections from "./sections/sections"
 
 interface ArticleProps {
   article: any
 }
+interface ArticleState {
+  isViewerOpen: boolean
```
Comment (eessex) [inline src/components/publishing/article.tsx]: very minor -- a bit syntactical if worded `viewerIsOpen`

File: src/components/publishing/sections/fullscreen_viewer/__test__/fullscreen_viewer.test.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  const viewer = renderer.create(<FullscreenViewer images={Images} show onClose={onClose} />).toJSON()
+  expect(viewer).toMatchSnapshot()
+})
+
+it("closes the viewer on ESC keydown", () => {
+  const onClose = jest.fn()
+  const viewer = shallow(<FullscreenViewer images={Images} show onClose={onClose} />)
+  viewer.simulate("keyDown", { keyCode: 27 })
+  expect(onClose.mock.calls.length).toBeGreaterThan(0)
+})
```
Comment (eessex) [inline src/components/publishing/sections/fullscreen_viewer/__test__/fullscreen_viewer.test.tsx]: nice tests

File: src/components/publishing/sections/fullscreen_viewer/slide.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+const Slide = props => {
+  const newProps = omit(props, ["section", "index", "total", "isCaptionOpen"])
+  const section = props.section
+  const src = section.url || section.image
+  return (
+    <div {...newProps}>
+      <SlideContainer>
+        <Title>{section.setTitle}</Title>
+        <Image src={src} />
```
Comment (eessex) [inline src/components/publishing/sections/fullscreen_viewer/slide.tsx]: This should include the caption as alt text for accessibility

File: src/components/publishing/sections/fullscreen_viewer/__test__/fullscreen_viewer.test.tsx (hunk 3 above)
Comment (orta) [reply src/components/publishing/sections/fullscreen_viewer/__test__/fullscreen_viewer.test.tsx]: @kanaabe you can change that to `expect(onClose).toBeCalled()`

Comment (orta) [reply src/components/publishing/sections/fullscreen_viewer/__test__/fullscreen_viewer.test.tsx]: Or `toBeCalledWith` also - https://facebook.github.io/jest/docs/en/expect.html#tohavebeencalled

File: src/components/publishing/sections/fullscreen_viewer/slide.tsx (hunk 4 above)
Comment (kanaabe) [reply src/components/publishing/sections/fullscreen_viewer/slide.tsx]: 👏 Best catch ever

**Comment by alloy** [comment]
@kanaabe @eessex FYI When adding a new dependency that does not ship with its own TypeScript typings (which is probably most of them), it’s worth it to check the [DefinitelyTyped repo](https://github.com/DefinitelyTyped/DefinitelyTyped) if others in the community have made typings for it.

In this instance that [appears to be the case](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-slick/index.d.ts). You can add those like so:

```
$ yarn add @types/react-slick --dev
```

**Pull request merged**